overlord/boot/firstboot.go

```diff
@@ -126,6 +136,78 @@ func populateStateFromSeed() error {
 	return ovld.Stop()
 }
 
+func importAssertionsFromSeed(st *state.State) error {
+	st.Lock()
+	defer st.Unlock()
+
+	assertSeedDir := filepath.Join(dirs.SnapSeedDir, "assertions")
+	dc, err := ioutil.ReadDir(assertSeedDir)
+	if err != nil {
+		return fmt.Errorf("cannot read assert seed dir: %s", err)
+	}
+
+	// FIXME: remove this check once asserts are mandatory
+	if len(dc) == 0 {
+		return nil
+	}
+
+	// collect
+	var modelAssertion *asserts.Model
+	bs := asserts.NewMemoryBackstore()
+	db, err := asserts.OpenDatabase(&asserts.DatabaseConfig{
```

Comment:
this is a bit strange, creating the db, but putting things into bs directly. What I would do is use bs
and db := assertstate.DB(st), then write retrieve in terms of just bs